src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1755,6 +1789,26 @@ private void emitNoResolvedType(NoType type, String defaultEmit) {
       }
     }
 
+    private String getDisplayNameFromType(JSType type) {
+      String displayName = type.getDisplayName();
+      // In partial mode, closure doesn't know the correct name of imported symbols, if the name
+      // matches one in the precomputed map, replace it with the original declared name
+      // The displayName can be of the form foo.bar, but the symbol that was goog required was just
+      // foo, so just replace the part of the display name before the first period
```

Comment:
I didn't follow this example too well, because there's only one period in the name.  Maybe make it instead describe foo.bar.baz so you can see the difference between the periods?

Reply:
Rewritten - take another look?